crates/wasmtime/src/values.rs

```diff
@@ -116,7 +129,14 @@ impl Val {
                     }))
                 }
             }
-            _ => unimplemented!("Val::read_value_from: {:?}", ty),
+            ValType::FuncRef => {
+                let func = ptr::read(p as *const *mut runtime::VMCallerCheckedAnyfunc);
+                if let Some(f) = NonNull::new(func) {
```

Comment:
Is `NonNull::new` here necessary?  This is related to my question above too I think.

I figured that `ref.null` would be implemented as returning the null singleton rather than a literal null pointer. (to make `table.set` a simple store instead of having a branch to store a non-null value)